contracts/test/MaliciousAccount.sol

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.12;
+import "../interfaces/IAccount.sol";
+import "../interfaces/IEntryPoint.sol";
+
+contract MaliciousAccount is IAccount {
```

Comment:
it is unclear from this test what is purpose, and how it is malicous.
specifically, I don't understand from the class (and from the test) how it "records" the original block basefee, in order to fail on the 2nd block.

Reply:
I'm extracting on-chain the block.basefee as you can see, and passing the expected basefee to compare with through maxPriorityFeePerGas parameter. It could be through any parameter, including nonce and signature.

Reply:
I cannot see: looking at the contract it doesn't make any sense, since you don't explain how it is used.